src/d3a/d3a_core/export.py

```diff
@@ -725,19 +741,21 @@ def add_plotly_graph_dataset(market_trades, market_slot):
         zero_point_dict = {"timestamp": [market_slot - GlobalConfig.tick_length],
                            "energy": [0.0]}
         # 1. accumulate data by buyer and seller:
-        for trade in market_trades:
-            trade_time = from_timestamp(trade["trade_time"], tz=TIME_ZONE)
-            seller = trade["seller"]
-            buyer = trade["buyer"]
-            energy = trade["energy"]
-            if seller not in seller_dict:
-                seller_dict[seller] = deepcopy(zero_point_dict)
-            if buyer not in buyer_dict:
-                buyer_dict[buyer] = deepcopy(zero_point_dict)
-            seller_dict[seller]["timestamp"].append(trade_time)
-            seller_dict[seller]["energy"].append(energy * ENERGY_SELLER_SIGN_PLOTS)
-            buyer_dict[buyer]["timestamp"].append(trade_time)
-            buyer_dict[buyer]["energy"].append(energy * ENERGY_BUYER_SIGN_PLOTS)
+        for stock_time, stocks in market_trades.items():
+            for stock in stocks:
```

Comment:
I dont think stock is the right naming here. I suggest to rename it to market_slots and to loop over the trades. so:
```suggestion
        for market_slot_time, market_slot_trades in market_trades.items():
            for trade in market_slot_trades:
```